cmd/protecodeExecuteScan_test.go

```diff
@@ -404,6 +404,16 @@ func TestGetTarName(t *testing.T) {
 			"",
 			"abc.tar",
 		},
+		"with sha ": {
+			"ppiper/cf-cli@sha256:c25dbacb9ab6e912afe0fe926d8f9d949c60adfe55d16778bde5941e6c37be11",
+			"",
+			"ppiper_cf-cli_c25dbacb9ab6e912afe0fe926d8f9d949c60adfe55d16778bde5941e6c37be11.tar",
+		},
+		"with sha+version ": {
+			"ppiper/cf-cli@sha256:c25dbacb9ab6e912afe0fe926d8f9d949c60adfe55d16778bde5941e6c37be11",
+			"verZion",
```

Comment:
This should fail though, simply because one cannot download with sha and version tag. But maybe that is of no concern of the `getTarName()` function, it should not even reach here. Anyway, I would expect for a version tag to be ignored, rather than appended, when using a sha digest.

Reply:
![Capture](https://user-images.githubusercontent.com/44555906/105168022-cdcce780-5b19-11eb-951e-0f9082a633ce.PNG)

At the current state - version tag isn't ignored and appended to the final string along with shadigest

Reply:
>At the current state - version tag isn't ignored and appended to the final string along with shadigest

I got that from the test. Would you care to code that? Extracting the version tag is not necessary in case of sha digest. Unless you say that adds unnecessary complications when this function should not be called in the first place if somebody tries to download a docker image with both sha and version.

Reply:
> > At the current state - version tag isn't ignored and appended to the final string along with shadigest
> 
> I got that from the test. Would you care to code that? Extracting the version tag is not necessary in case of sha digest. Unless you say that adds unnecessary complications when this function should not be called in the first place if somebody tries to download a docker image with both sha and version.

Not positive what needs to be done here? What exactly needs coding here? 
Version is just a string which appends to the dockerImage, if the user has set it and it is for mere identification in protecode.

Reply:
Maybe we can rewrite all the above unit-tests for protecodeExecuteScan_test.go?